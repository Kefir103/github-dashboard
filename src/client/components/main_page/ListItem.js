import React, { Component } from 'react';
import history from '../../history';

export default function ListItem(props) {
    return (
        <div
            className={'list-item'}
            onClick={() => {
                history.push(`/repos/${props.repository.name}`, {...props.repository});
            }}>
            <h4>
                {props.repository.name} ({props.repository.stargazers_count} звезд)
            </h4>
            <p>Последнее изменение: {new Date(props.repository.pushed_at).toLocaleString()}</p>
            <p>URL: <a href={props.repository.html_url}>{props.repository.html_url}</a></p>
        </div>
    );
}
