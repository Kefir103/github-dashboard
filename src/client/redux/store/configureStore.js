import { createStore, applyMiddleware, compose } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension';
import thunk from 'redux-thunk';
import { rootReducer } from '../reducers/rootReducer';

let store;

export default function configureStore(initialState) {
    if (store) {
        return store;
    }

    store =
        process.env.NODE_ENV === 'production'
            ? createStore(rootReducer, initialState, compose(applyMiddleware(thunk)))
            : createStore(rootReducer, initialState, composeWithDevTools(applyMiddleware(thunk)));

    return store;
}
