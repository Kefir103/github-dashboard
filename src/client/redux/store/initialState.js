export const initialState = {
    appStatus: {
        isLoading: true,
    },
    repository: {
        repos: [],
        totalCount: 0,
        contributors: [],
        currentRepository: {},
    },
    filter: {
        searchText: '',
        currentPage: 1,
    },
};
