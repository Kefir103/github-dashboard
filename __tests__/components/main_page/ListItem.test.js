import React from 'react';
import configureMockStore from 'redux-mock-store';
import { initialState } from '../../../src/client/redux/store/initialState';
import ListItem from '../../../src/client/components/main_page/ListItem';
import thunk from 'redux-thunk';
import history from '../../../src/client/history';
import { shallow, render, mount } from 'enzyme';

const mockStore = configureMockStore([thunk]);

describe('<ListItem/>', () => {
    let store;
    beforeEach(() => {
        store = mockStore({ ...initialState });
    });
    test('should be rendered correctly', () => {
        const repository = {
            name: 'name',
            stargazers_count: 1,
            pushed_at: '2005-08-09T18:31:42',
            html_url: 'http://localhost:3000',
        };
        const wrapper = shallow(
            <ListItem repository={repository} store={store} history={history} />
        );
        const component = render(wrapper);
        expect(component).toMatchSnapshot();
    });
    test('should change repository correctly', () => {
        const repository = {
            name: 'name',
            stargazers_count: 1,
            pushed_at: '2005-08-09T18:31:42',
            html_url: 'http://localhost:3000',
        };
        const wrapper = mount(<ListItem repository={repository} store={store} history={history} />);
        wrapper.find('.list-item').simulate('click');
        expect(wrapper.props().history.location.pathname).toEqual(`/repos/${repository.name}`);
        expect(wrapper.props().store.getActions()[0].payload.repository).toEqual(repository);
    });
});
