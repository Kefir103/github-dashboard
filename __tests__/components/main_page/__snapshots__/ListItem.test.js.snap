// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListItem/> should be rendered correctly 1`] = `
<div
  class="list-item"
>
  <h4>
    name (1 звезда)
  </h4>
  <p>
    Последнее изменение: 2005-8-9 18:31:42
  </p>
  <p>
    URL: 
    <a
      href="http://localhost:3000"
    >
      http://localhost:3000
    </a>
  </p>
</div>
`;
